c/progress/manager.h

```diff
@@ -43,9 +44,10 @@ class progress_manager {
     // while more and more top-level tasks are received.
     progress_bar* pbar;
     std::stack<work*> tasks;
+    std::mutex mutex;
 
   public:
-    void update_view() const;
+    void update_view();
```

Comment:
`update_view()` is still conceptually a `const` method: it isn't supposed to change the state of the `progress_manager` instance. So you could either declare `mutex` as `mutable` (I intend all my puns), or even make it a static file-level variable.

Reply:
If you want to keep this method `const`, I guess it is better to declare `mutex` as mutable and keep it as a part of the class for the purpose of encapsulation.